src/XGBoostSharp/lib/DMatrix.cs

```diff
@@ -17,12 +17,28 @@ public float[] Label
         set { SetFloatInfo(Fields.label, value); }
     }
 
-    public DMatrix(float[][] data, float[] labels = null)
-        : this(Flatten2DArray(data), unchecked((ulong)data.Length), unchecked((ulong)data[0].Length), labels)
+    /// <summary>
+    /// Creates a DMatrix from a 2D array of float values with optional labels.
+    /// </summary>
+    /// <param name="data">The 2D array containing the feature data.</param>
+    /// <param name="labels">Optional array of labels corresponding to the data rows. Default is null.</param>
+    /// <param name="m_missing">Value to be treated as missing in the dataset. Default is float.NaN.</param>
+    /// <exception cref="DllFailException">Thrown when the native XGBoost library encounters an error during matrix creation.</exception>
+    public DMatrix(float[][] data, float[] labels = null, float m_missing = DefaultMissing)
```

Comment:
@MiWeiss rename argument from `m_missing` to `missing`. same in the other constructor.